From 5d55dcaf064a1e3961aa906398d66419d1e849ad Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Tue, 28 Apr 2020 16:08:33 +0200
Subject: [PATCH] sja1105 module Makefile didn't respect install dirs

Yocto passes in the MODLIB and INSTALL_FW_PATH variables.
If the recipe doesn't respect them, modules will be installed
in wrong places depending on the rootfs config.
Specifically DISTRO_FEATURES "usrmerge" failed.

Signed-off-by: Heinz Wrobel <Heinz.Wrobel@nxp.com>
---
 Makefile | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 6c006e0..93b48b6 100644
--- a/Makefile
+++ b/Makefile
@@ -8,6 +8,8 @@ MYPLATFORM=bbmini
 #SPI_BPW_MSG
 #NR_CFG_BLOCKS
 
+INSTALL_FW_PATH=$(INSTALL_DIR)/lib/firmware
+MODLIB=$(INSTALL_DIR)/lib/modules/$(KERNEL_VERSION)
 
 ifeq ($(MYPLATFORM), imx)
 	# i.MX
@@ -81,16 +83,16 @@ clean: firmware_clean
 
 modules_install: firmware_install
 	$(eval KERNEL_VERSION := $(shell cat $(KERNELDIR)/include/config/kernel.release))
-	install -D sja1105pqrs.ko $(INSTALL_DIR)/lib/modules/$(KERNEL_VERSION)/kernel/drivers/spi/sja1105pqrs.ko
+	install -D sja1105pqrs.ko $(MODLIB)/kernel/drivers/spi/sja1105pqrs.ko
 
 firmware:
 	$(MAKE) -C $(FIRMWARE_DIR) BOARD=$(MYPLATFORM) NR_SWITCHES=$(NUMBER_SWITCHES)
 
 firmware_install:
-	@mkdir -p $(INSTALL_DIR)/lib/firmware/
+	@mkdir -p $(INSTALL_FW_PATH)
 	@$(foreach SWITCH, $(shell seq 1 $(NUMBER_SWITCHES) ), \
 		install -D $(FIRMWARE_DIR)/sja1105p_$(SWITCH)-$(NUMBER_SWITCHES)_cfg.bin \
-			$(INSTALL_DIR)/lib/firmware/;)
+			$(INSTALL_FW_PATH);)
 
 firmware_clean:
 	$(MAKE) -C $(FIRMWARE_DIR) clean
-- 
2.17.1

